src/components/Table/TableHead/TableHead.jsx

```diff
@@ -40,6 +40,7 @@ const { iotPrefix } = settings;
 const { TableHead: CarbonTableHead, TableRow, TableExpandHeader } = DataTable;
 
 const propTypes = {
+  id: PropTypes.string.isRequired,
```

Comment:
I think we usually follow the convention that internal table components take a `tableId` instead of `id`
```suggestion
  tableId: PropTypes.string.isRequired,
```